tests/storage-vm

```diff
@@ -242,6 +242,49 @@ for poolDriver in $poolDriverList; do
         echo "==> Deleting VM"
         lxc delete -f v1
 
+        if hasNeededAPIExtension disk_state_created; then
+                # Check that we don't remove an existing target directory when unmounting a disk device if the original dir hasn't been created by LXD
+                echo "==> Avoid deleting existing target directories not created by LXD when unmounting disks."
+                mkdir -p "/tmp/lxd-test-${poolName}/empty-dir"
+                lxc launch "${TEST_IMG:-ubuntu-minimal-daily:22.04}" v1 --vm -s "${poolName}"
+                waitInstanceReady v1
+
+                ! lxc exec v1 -- mkdir /opt || false # The directory already exists and is not created by LXD.
+                lxc config device add v1 empty-dir disk source="/tmp/lxd-test-${poolName}/empty-dir" path=/opt
+                # Check that no `volatile.<deviceName>.last_state.created` has been set.
+                if [ -n "$(lxc config get v1 volatile.empty-dir.last_state.created)"]; then
+                        echo "==> volatile.<deviceName>.last_state.created should not be set"
+                        false
+                fi
+
+                lxc config device remove v1 empty-dir
+                lxc exec v1 -- stat /opt # /opt is not removed as it was not created by LXD.
+
+                lxc exec v1 -- mkdir -p /tmp/empty-dir # Create a new directory to check that it is not removed when unmounting the disk.
```

Comment:
```suggestion
                lxc exec v1 -- mkdir /tmp/empty-dir # Create a new directory to check that it is not removed when unmounting the disk.
```

`/tmp` has to be there, otherwise we have other problems.

Reply:
fixed